vllm/forward_context.py

```diff
@@ -61,7 +64,8 @@ def get_forward_context() -> ForwardContext:
 def set_forward_context(attn_metadata: Any,
```

Comment:
suggestion not must have to: can we make a new function - set_forward_context_dp and redirect from set_forward_context?
I think current change is quite intrusive, and it might be very difficult to for habana team to do rebase, Isolate the DP codes in a separate function might be helpful.